README.md

```diff
@@ -1,70 +1,291 @@
-# deja-fu
+# Déjà Fu
 
 <!-- badges -->
 [![cljdoc badge](https://cljdoc.org/badge/com.lambdaisland/deja-fu)](https://cljdoc.org/d/com.lambdaisland/deja-fu) [![Clojars Project](https://img.shields.io/clojars/v/com.lambdaisland/deja-fu.svg)](https://clojars.org/com.lambdaisland/deja-fu)
 <!-- /badges -->
 
 Lightweight ClojureScript local time/date library
 
-While this library is relatively new, it has a battery of tests and has been used in production code.
-
-## Features
+> A martial art in which the user's limbs move in time as well as space,
+> allowing them to go back in time and punch the opponent repeatedly so that the
+> opponent feels the effects of past blows while standing in front of the
+> apparently motionless [Déjà > Fu](https://wiki.lspace.org/mediawiki/D%C3%A9j%C3%A0_Fu) practitioner. Similar
+> tricks can be used to render the opponent's attacks ineffective or otherwise
+> incapacitate him.
+>
+> It is best described as "the feeling that you have been kicked in the head this way before".
+>
+> — Terry Pratchet
 
 <!-- installation -->
 ## Installation
 
-To use the latest release, add the following to your `deps.edn` ([Clojure CLI](https://clojure.org/guides/deps_and_cli))
+Deja-fu is still in alpha, for now you can use it as a git dependency from `deps.edn`.
+
+<!-- /installation -->
 
+## Rationale
+
+This library is the result of a particular set of insights and design
+constraints. The main insight is that it is valuable to have distinct types for
+distinct concepts like a "calendar date" vs "wall time" vs a "date-time /
+timestamp". On the JVM we have known this for some time, thanks to Joda-Time and
+later the java.time API (aka JSR-310).
+
+The equivalent in the JavaScript/ClojureScript world is js-joda, which is used
+by the ClojureScript version of [juxt/tick](https://github.com/juxt/tick), or
+the upcoming [TC39 Temporal API](https://github.com/tc39/proposal-temporal).
+Eventually we can expect Temporal to become part of browsers and other JS
+runtimes, but until then a hefty polyfill is needed, significantly blowing up
+build size. The same is true of js-joda.
+
+For applications where dealing with time is not enough of their core business to
+justify these large dependencies, a more lightweight alternative is needed. This
+is where deja-fu comes in. It builds upon the Google Closure library's
+`goog.date.Date` and `goog.date.DateTime`, and adds a third "time" type, thus
+providing representations for three concepts.
+
+- a date
+- a time without time zone
+- a date+time without time zone
```

Comment:
It's a little confusing which of these are Google Closure types and which are Deja-Fu types.